src/kratos.py

```diff
@@ -117,16 +118,62 @@ def recover_password_with_code(self, identity_id: str, expires_in: str = "1h") -
 
         return r.json()
 
-    def recover_password_with_link(self, identity_id: str, expires_in: str = "1h") -> Dict:
-        """Create a magic link for recovering an identity's password."""
-        url = join(self.kratos_admin_url, "admin/recovery/link")
-        data = {"identity_id": identity_id, "expires_in": expires_in}
-
-        r = requests.post(url, json=data)
+    def reset_password(self, identity_id: str, password: str) -> Dict:
+        """Set identity's password to a provided value."""
+        identity = self.get_identity(identity_id)
+        traits = identity.get("traits")
+        schema_id = identity.get("schema_id")
+        state = identity.get("state")
+
+        credentials = {
+            "password": {
+                "config": {
+                    "hashed_password": bcrypt.hashpw(password.encode("utf-8"), bcrypt.gensalt()).decode("utf-8")
```

Comment:
Bcrypt is the [default](https://www.ory.sh/docs/self-hosted/kratos/configuration/password#bcrypt) algorithm for hashing, let me know it should be changed